pandas/_libs/groupby_helper.pxi.in

```diff
@@ -588,7 +588,16 @@ def group_rank_{{name}}(ndarray[float64_t, ndim=2] out,
                 if out[i, 0] != out[i, 0] or out[i, 0] == NAN:
                     out[i, 0] = NAN
                 else:
-                    out[i, 0] = out[i, 0] / grp_sizes[i, 0]
+                    with gil:
```

Comment:
Instead of going back to the gil can't you just check if the denominator is zero?